directory.tm.json

```diff
@@ -9,9 +9,12 @@
     ],
     "title": "Thing Description Directory (TDD) Thing Model",
     "version": {
-        "model": "1.0.0-alpha"
+        "model": "1.0.0-beta"
     },
     "base": "{{DIRECTORY_BASE_URL}}",
+    "tm:required": [
+        "#/properties/things"
+    ],
```

Comment:
The `things` property has [two forms](https://github.com/w3c/wot-discovery/blob/51c051368fc0015d3173164b4ae9ed0a0475a169/directory.tm.json#L55-L86). One is for array payload which is mandatory, another is for the collection payload which is optional. To define this required form, the TM should support setting of a form to required using a nested `required: true` field. It currently doesn't.

I am setting the whole property to required, assuming that it means at least one form should offer this interaction. I can also set to `#/properties/things/forms/0` but that means we are forcing the order of items in the forms array.

This was discussed in https://github.com/w3c/wot-discovery/pull/253#issuecomment-1004594898

Reply:
TBH, I think that for now having only `things` property required is fine. I know that it is clearly depicting the whole picture but we have normative statements that cover that part. I would wait for the TD spec to catch up and deal with this kind of issue (I would say that the issue here is much more complex than just requiring that a derived TD must have a specific form).

Reply:
OK for now with this resolution, but I'm wondering now if we should not have two different interactions here.  The expectation in TDs are that different forms are somehow equivalent, not for different purposes.  The scripting API, for example, also assumes that any form in an interaction will do the same thing as any other.

Reply:
They are equivalent ways of getting the TDs. The only difference is payload and header formats.

Reply:
Discussion: Farshids thinks this is broken (maybe messed up as part of an earlier PR) and needs to be fixed.  McCool thinks at the very least that putting forms distinguished by different contentTypes in a single interaction is inconvenient for Consumers.  Farshid will create an issue and will investigate.